coalib/bearlib/languages/definitions/Haskell.py

```diff
@@ -0,0 +1,18 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Haskell:
+    extensions = '.hs'
+    versions = 1.0, 1.1, 1.2, 1.3, 1.4
+    special_chars = '!#$%&*+./<=>?@\\^|-~'
+    comment_delimiters = '-- |',
+    multiline_comment_delimiters = {'-|': '-'}
+    grouping_expressions = {'$'}
```

Comment:
I've seen several attributes here like `grouping_expressions`, `string_accessing_n-th_character`, but not seen in any other language definition file or Language.py. Could you explain a bit about what they are for?